I can't make file when I follow the installation guide

I want  to use zgrab2  to do some secondary development ,so I follow the installation guide to install the zgrab2.
when I try to make file , it outputs the error 
```
cd cmd/zgrab2 && go build && cd ../..
go: downloading gopkg.in/yaml.v2 v2.2.2
go: downloading gopkg.in/mgo.v2 v2.0.0-20190816093944-a6b53ec6cb22
go: updates to go.sum needed, disabled by -mod=readonly
Makefile:24: recipe for target 'zgrab2' failed
make: *** [zgrab2] Error 1
```

I don't know how to solve this problem. I use go mod, so I run 
```
cd  /zmap$ cd pkg/mod/github.com/zmap/zgrab2@v0.1.7/
make
```
go version is 1.15.5
please provide me some advice, thank u

--- a/README.md
+++ b/README.md
@@ -25,31 +25,28 @@ Replace `/path/to/your/config.ini` with the path to your configuration file on t
 
 ### Building from Source
 
-For Go 1.17 and later you must build from source:
+ZGrab2 requires Go 1.23 or later to build from source.
 
 ```shell
 git clone https://github.com/zmap/zgrab2.git
 cd zgrab2
 make
-./zgrab2
+./zgrab2 http --help # to see the http module's help message
 ```
 
+Starting in Go 1.21, Go added [support](https://go.dev/doc/toolchain) for auto-downloading the appropriate toolchain for building a given module.
 
-For Go 1.16 and below you can install via go get:
-
-You will need to have a valid `$GOPATH` set up, for more information about `$GOPATH`, see https://golang.org/doc/code.html.
-
-Once you have a working `$GOPATH`, run:
+This will let you build ZGrab2 using Go 1.21.X or 1.22.X without needing to manually install another version.
 
 ```shell
-go get github.com/zmap/zgrab2
-```
+go version
+$ go version go1.21.13 linux/arm64
 
-This will install zgrab under `$GOPATH/src/github.com/zmap/zgrab2`
-
-```shell
-cd $GOPATH/src/github.com/zmap/zgrab2
-make
+export GOTOOLCHAIN=auto
+git clone https://github.com/zmap/zgrab2.git
+cd zgrab2
+make # Go will download the required 1.23 toolchain automatically
+./zgrab2 http --help # to see the http module's help message
 ```
 
 ## Single Module Usage 

